usaspending_api/etl/management/commands/load_multiple_submissions.py

```diff
@@ -1,126 +1,139 @@
-                cgac = next_broker_sub[2]
-                frec = next_broker_sub[3]
-
-                api_cursor.execute(
-                    "SELECT update_date FROM submission_attributes WHERE submission_id = {}".format(submission_id)
-                )
-
-                if api_cursor.rowcount:
-                    most_recently_loaded_date = api_cursor.fetchone()[0].replace(tzinfo=pytz.UTC)
-                else:
-                    most_recently_loaded_date = datetime(2000, 1, 1).replace(tzinfo=pytz.UTC)
-
-                if frec:
-                    broker_cursor.execute("SELECT agency_name FROM frec WHERE frec_code = '{}'".format(frec))
-                else:
-                    broker_cursor.execute("SELECT agency_name FROM cgac WHERE cgac_code = '{}'".format(cgac))
-
-                agency_name = broker_cursor.fetchone()[0]
-
-                if certify_date > most_recently_loaded_date:
-                    missing_submissions.append((submission_id, agency_name, certify_date, most_recently_loaded_date))
-            except Exception:
-                logger.exception("Submission ID {} failed in pull from broker".format(submission_id))
-                failed_submissions.append(str(submission_id))
+    def handle(self, *args, **options):
 
-        if len(missing_submissions):
-            logger.info("Total missing submissions: {}".format(len(missing_submissions)))
-            logger.info("-----------------------------------")
-            for submission in missing_submissions:
-                logger.info(
-                    "Submission ID {} ({})\tCertified: {}".format(submission[0], submission[1], submission[2].date())
-                )
-            logger.info("-----------------------------------")
+        if options["submission_ids"]:
+            submission_ids = options["submission_ids"]
         else:
-            logger.info("No DABS to load")
-
-        # Data modification happens here, if you don't flag '--safe'
-        # The submission loader is atomic, so one of these failing will not affect subsequent submissions
-        if options["safe"]:
-            logger.info("Exiting script before data load occurs in accordance with the --safe flag.")
+            submission_ids = self.get_incremental_submission_ids()
+
+        if submission_ids:
+            msg = f"{len(submission_ids):,} submissions will be created or updated"
+            if len(submission_ids) <= 1000:
+                logger.info(f"The following {msg}: {submission_ids}")
+            else:
+                logger.info(f"{msg}.")
+            if options["list_ids_only"]:
+                logger.info("Exiting script before data load occurs in accordance with the --list-ids-only flag.")
+                return
+        else:
+            logger.info("There are no new or updated submissions to load.")
             return
-
-        for submission in missing_submissions:
-            submission_id = submission[0]
+        print("oh dear")
```

Comment:
💯

Reply:
that

is

embarrassing

Reply:
Could have been worse. Thankfully you don't print your SSN when debugging

Reply:
Maybe if I did I'd make fewer oopsies.